components/gitpod-db/src/typeorm/migration/1675744891502-WorkspaceOrganizationId.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+const table = "d_b_workspace";
+const column = "organizationId";
+const orgIndex = "idx_organizationId";
+
+export class WorkspaceOrganizationId1675744891502 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, table, column))) {
+            await queryRunner.query(
+                `ALTER TABLE ${table} ADD COLUMN ${column} varchar(36), ALGORITHM=INPLACE, LOCK=NONE`,
+            );
+
+            // Update the data in chunks so we don't lock the table for too long.
+            // Tested the corresponding SELECT queries on prod failover and the slow ones take about max 15s.
+            const dates = [
+                "2023-02-03",
```

Comment:
How are we going to handle the case when the migration may be executed at some point in the future?

Reply:
You mean on prod? We can't rerun it in prod (incl. the addition of the column).

Reply:
This is the same for all migrations we have added in code but then manually did the change in prod.

Reply:
Maybe we should not implement the `down` logic but make the `up` logic backward compatible (which it usually is)

Reply:
This migration will execute on ANY Gitpod installation that gets upgraded. We don't know when it gets upgraded. The code here only handles data up to this date. But other installations do not conform to the same points in time (presumably this date is determined out of SaaS).

Reply:
It updates everything that exists at that point in the DB. But, yes, the code that adds the organizationId to new workspaces going forward would need to be deployed as well. That's why I initially had these changes in one PR.